Make the install page reasonable (closes #256).

dbuenzli opened the pull request:
The only thing I didn't clean are the Windows instructions since I don't know enough about what are resonable instructions.

I didn't try to build the website. A basic rendering of the new page can be seen here: 
https://github.com/dbuenzli/ocaml.org/blob/master/site/docs/install.md

avsm commented on site/docs/install.md:
```diff
@@ -8,255 +8,144 @@ The latest version of OCaml is **{{! get LATEST_OCAML_VERSION !}}**.
+Other Debian packages related to OCaml are
+[listed here](http://packages.debian.org/search?keywords=ocaml&searchon=all&suite=testing&section=all).
+They are made available through
+[The OCaml debian packaging policy](http://pkg-ocaml-maint.alioth.debian.org/ocaml_packaging_policy.html/index.html).
```
It's the "Debian OCaml packaging policy" I think

avsm commented on site/docs/install.md:
```diff
@@ -8,255 +8,144 @@ The latest version of OCaml is **{{! get LATEST_OCAML_VERSION !}}**.
-You can view the list of OCaml packages in Ubuntu repositories
-[here](http://packages.ubuntu.com/search?keywords=ocaml).
+[Ubuntu](http://www.ubuntu.com/) is a Debian derivative so the
+[Debian instructions](#Debian) are applicable. Other Ubuntu packages related
```
How are the Debian instructions applicable?  The packages don't directly work (or at least, it's not recommended to mix binary packages between Ubuntu and Debian).  It would be shorter to recommend source installation via OPAM perhaps.

FWIW, I do have Debian Jessie backport Dockerfiles here: https://github.com/avsm/debian-jessie-backports-opam that generate Debs of OPAM.  I could look into compiling regular releases of these for Ubuntu as well.

dbuenzli replied:
> How are the Debian instructions applicable? The packages don't directly work (or at least, it's not recommended to mix binary packages between Ubuntu and Debian).

This is what was written on the previous page. But WDYM, why wouldn't `apt-get install ocaml` work ?

dbuenzli replied:
In any case I'll simply repeat the instructions since the person who wants to install ocaml doeesn't give a shit that Ubuntu is a Debian derivative.

avsm replied:
Yes indeed, the background information on Ubuntu is pretty irrelevant here

avsm commented on site/docs/install.md:
```diff
@@ -8,255 +8,144 @@ The latest version of OCaml is **{{! get LATEST_OCAML_VERSION !}}**.
+is to use « Bash on Windows » and install OCaml as on Ubuntu. Main
+OCaml versions and OPAM work. You can use OCaml as you are on
+Ubuntu. The disadvantage of this method is you don't build native
+Windows binaries but Linux binaries.
```
There is also http://protz.github.io/ocaml-installer/ which is 4.02.3 (unlike OCPWin, which still seems to be 4.01.0)

dbuenzli replied:
This is the one that is linked under "Official OCaml distribution in fact".

avsm replied:
It's not really clear to me what "official" means in this capacity, since it's not generated by the INRIA release manager (@damiendoligez).  It's also a major release out of date -- @msprotz, any thoughts on how we should refer to your distribution in the Windows install section?

msprotz replied:
Nope... maybe "trusted", or "partially maintained"? I'm in kind of a best effort situation right now... we talk about it at the ML workshop. I'm around Nara.

avsm commented on site/docs/install.md:
```diff
@@ -305,279 +194,23 @@ installers and zip archives are provided for OCaml 4.01.0 and 4.02.1. Some addit
+* An
+  [OCaml toplevel](http://ocsigen.org/js_of_ocaml/dev/files/toplevel/index.html)
+  compiled by the [`js_of_ocaml`](http://ocsigen.org/js_of_ocaml/)
+  project.
```
We could also link to iocamljs here: https://andrewray.github.io/iocamljs/

avsm commented on site/docs/install.md:
````diff
@@ -8,255 +8,144 @@ The latest version of OCaml is **{{! get LATEST_OCAML_VERSION !}}**.
+pkg_add -r ocaml-nox11 # If you don't want X11 support 
+pkg_add -r ocaml
+```
 
````
OpenBSD has also had support for ages.  The instructions are:

## [OpenBSD](https://www.openbsd.org)

```bash
pkg_add ocaml opam
```

dbuenzli replied:
Will add this.

dbuenzli commented:
Amended the commit with @avsm comments. 

Finally went through the windows section aswell: Removed a lot of things since the details mentioned can be obtained at the link endpoints. I also added a link to @fdopen's experimental opam port since this is what worked best for me at some point. I removed the link to `wodi` since the page says that it's no longer supported. 

I'm sure the windows stuff could still be improved and trimmed. I somehow doubt about the "Use « Bash on Windows » and install OCaml as on Ubuntu.". I left it since I don't know what it means, but I highly suspect that bash on windows doesn't give you `apt-get`...

dbuenzli commented:
Is there still something that needs to be done for this to be merged ? The current install page makes me physically ill.

pw374 commented:
Thanks.
